gh-126108: Fix potential null pointer dereference in PySys_AddWarnOptionUnicode (#126118)
python/cpython: Misc/NEWS.d/next/Security/2024-10-29-09-15-10.gh-issue-126108.eTIjHY.rst

> Fix potential null pointer dereference in PySys_AddWarnOptionUnicode

Reviewer: ```suggestion
Fix a possible ``NULL` pointer dereference in :c:func:`!PySys_AddWarnOptionUnicode`.
```

FTR: The function is technically deprecated but is still present.
